### ISSUE TITLE
macro-error: "unexpected token" for test_case attribute with semicolon syntax

### ISSUE THREAD

**Issue by nazmulidris** [rust-lang/rust-analyzer#21216]
## rust-analyzer version
0.3.2702-standalone (VS Code extension, both release and pre-release)

## rustc version
rustc 1.93.0-nightly (3ff30e7 2025-11-29)

## Relevant settings
Default settings, proc-macros enabled.

## Problem
rust-analyzer reports `macro-error: unexpected token` on `#[test_case]` attributes that use the semicolon syntax for naming test cases.

The code compiles and runs correctly with `cargo test`.

## Minimal reproduction

```rust
// Cargo.toml: test-case = "3"

use test_case::test_case;

#[test_case(80, 1  ; "80 cols = 1 row boundary")]
#[test_case(160, 2 ; "160 cols = 2 row boundary")]
fn my_test(input: u16, expected: u16) {
    assert_eq!(input / 80, expected);
}
```

## Expected behavior
No diagnostic errors — the `test_case` proc-macro should expand successfully.

## Actual behavior
rust-analyzer shows `macro-error: unexpected token` on the first `#[test_case]` attribute in each stacked group. Subsequent attributes in the same group don't show errors.

Additionally, because the macro fails to expand, all symbols inside the test function show as "unknown type" (no type inference).

## Notes
- The semicolon syntax `(args ; "name")` is documented `test_case` syntax for naming test cases
- Only the **first** `#[test_case]` in a stacked group shows the error
- `cargo check` reports no errors
- The issue appears in VS Code, VS Code Insiders, and VSCodium with both release and pre-release extensions

## Workaround
```json
"rust-analyzer.diagnostics.disabled": ["macro-error"]
```

**Comment by ChayimFriedman2** [comment]
The macro input gets wrapped in an extra `Group`. CC @Veykril this is probably your work.

**Comment by Veykril** [comment]
Can you try the latest nightly rust? This should be fixed there

**Comment by nazmulidris** [comment]
@Veykril I just upgraded to the latest nightly toolchain and the issues are different (but still not resolved). Here's a screenshot:

<img width="2487" height="1483" alt="Image" src="https://github.com/user-attachments/assets/cccc44a6-ea8a-4cc5-a0c4-f76dae6553c4" />

I tried it on vscode, code-insiders, and vscodium. They all report the same diagnostic errors from rust-analyzer.

Here are all the versions:
- rust-analyzer extension: 0.3.2702
- Rust [toolchain] channel = "nightly-2025-12-07" (latest nightly)
- Here's the code that triggers this diagnostic error: https://github.com/r3bl-org/r3bl-open-core/blob/0e9c715bd53a2be7b7d3cb9f466dcf4bf199ba8e/tui/src/readline_async/readline_async_impl/line_state/cursor.rs#L284C1-L287C55

**Comment by Veykril** [comment]
Yea noticed that as well earlier, https://github.com/rust-lang/rust-analyzer/pull/21223 will fix that. Tomorrows r-a release should have this fixed then

**Comment by ChayimFriedman2** [comment]
Should be fixed now.

**Comment by nazmulidris** [comment]
@ChayimFriedman2 Thank you! I can confirm it is working for me!

### `cursor.rs` - No longer shows error:
  <img width="2002" height="523" alt="Image" src="https://github.com/user-attachments/assets/59fc84e5-4fb2-45d9-9197-55b05716269b" />

### `settings.json` - I removed the workaround
  <img width="2002" height="523" alt="Image" src="https://github.com/user-attachments/assets/70a53dd0-87b8-492c-819b-e48e065f17c2" />

**Comment by wijonja** [comment]
confirmed, working for me also
